coding/trie.hpp

```diff
@@ -79,4 +83,19 @@ void ForEachRef(Iterator<TValue> const & iter, F & f, TString const & s)
   }
 }
 
-}  // namespace Trie
+template <typename TValueList, typename TF, typename TString>
+void ForEachRefWithValues(Iterator<TValueList> const & iter, TF && f, TString const & s)
```

Comment:
SearchTokensCollector in generator/dumper.cpp did not depend on values' type.
Now it does (while unnecessary, this may be viewed as an additional type check).
done